### ISSUE TITLE
SyntaxError: Cannot use import statement outside a module

### ISSUE THREAD

**Issue by wlievens** [StrandedKitty/straight-skeleton#2]
I'm trying to use this package in my TypeScript project, but I get an following error when I import something.

My import code:
```
import SkeletonBuilder from 'straight-skeleton';
```

The error:
```
import { List } from "./Utils";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    at Module._compile (node:internal/modules/cjs/loader:1067:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Object.require.extensions.<computed> [as .js] (C:\workspace\griffin\griffin-renderer\node_modules\ts-node\src\index.ts:1608:43)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\workspace\griffin\griffin-renderer\src\geometry.ts:2:1)
```

For reference in case it matters, this is my tsconfig.json:

```
{
  "extends": "./node_modules/gts/tsconfig-google.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "build",
    "lib": [
      "es2016",
      "es2021",
      "dom"
    ]
  },
  "include": [
    "src/**/*.ts",
    "test/**/*.ts",
    "src/examples/**/*.ts"
  ]
}
```

**Comment by StrandedKitty** [comment]
Hi!
Are you using NodeJS? Because the error looks like it was produced by NodeJS. Maybe the problem is that you don't have `"type": "module"` in your package.json file?

**Comment by wlievens** [comment]
Yes, I'm running from NodeJS indeed. I'm new to NodeJS so I didn't know I had to do that, I'll try it right away, thanks.

**Comment by wlievens** [comment]
I don't seem to be able to convert my project setup to `"type": "module"`, I get all sorts of issues that I don't understand. I run my code with ts-node, in case that matters. Is it really not possible to use this package with CommonJS imports? If works for all my other dependencies.

**Comment by StrandedKitty** [comment]
Hmm, I wonder if I should just set `"module": "commonjs"` in the tsconfig file... It shouldn't break anything I think. Honestly, I don't know why it's currently set to `esnext`, I guess I never thought anyone would use it in an environment that doesn't support modern modules.

**Comment by wlievens** [comment]
I made some progress with converting my project to use type:module but I'm not there yet. Will keep you posted :-)

**Comment by StrandedKitty** [comment]
I've published a new version (1.1.0) that outputs a commonjs module. Syntax for importing the library using the `import` keyword changed too (because now it needs to be compatible with the commonjs build), so now it can be imported like this:

`import {SkeletonBuilder} from 'straight-skeleton';` or 
`const {SkeletonBuilder} = require('straight-skeleton');`

**Comment by wlievens** [comment]
I got your new build to work, thanks!

I played with it and did see a funky result, I'll post another issue for that.
